java/memory/memory-core/src/main/java/org/apache/arrow/memory/ArrowBuf.java

```diff
@@ -346,6 +347,29 @@ public void setLong(long index, long value) {
     MemoryUtil.UNSAFE.putLong(addr(index), value);
   }
 
+  /**
+   * Get float16 value stored at a particular index in the
+   * underlying memory chunk this ArrowBuf has access to.
+   * @param index index (0 based relative to this ArrowBuf)
+   *              where the value will be read from
+   * @return 4 byte float value
+   */
+  public short getFloat16(long index) {
```

Comment:
Given that `setFloat16` takes a float, by symmetry this method should probably return a float too.
@danepitkin WDYT?

Reply:
Casting to float will involve a conversion to a 32 bit type. I think it would be nice to provide set/get operations on short as the core APIs and set/get operations on float as helper APIs. Thoughts?

Reply:
Well, `getShort` and `setShort` already seems provided?

Reply:
heh, good point. 

I agree it makes sense to change this to float.

Reply:
Changed